gnosis/safe-react
# fix: initialise web3 and load contracts in middleware

## What it solves
Resolves #3722

## How this PR fixes it
`setWeb3` and the instantiation of contracts has been moved to the provider middleware (as well as `resetWeb3`).

As we enable the pairing module on load, onboard's wallet subscription fires by default, meaning that the `updateProviderWallet` action is fired. Within that subscription we were saving the provider: a WalletConnect session that isn't connected (the QR code not scanned). This was causing issues retrieving the spending limit contract and preventing the Safe from loading.

## How to test it
Open a Safe with the spending limit module enabled (e.g. `/app/eth:0xfF501B324DC6d78dC9F983f140B9211c3EdB4dc7/balances`) and the wallet disconnected. The Safe should load without issues: balances, transactions, etc.

This fix affects the instantiation of the web3 object/contract loading and thus the transaction interaction must also be tested.
